contracts/middle-layer/resource-mirror/AdditionalBucketHub.sol

```diff
@@ -125,33 +148,31 @@ contract AdditionalBucketHub is BucketStorage, GnfdAccessControl {
 
         // check package queue
         if (extraData.failureHandleStrategy == FailureHandleStrategy.BlockOnFail) {
-            require(retryQueue[msg.sender].empty(), "retry queue is not empty");
+            require(retryQueue[sender].empty(), "retry queue is not empty");
         }
 
         // check authorization
-        address owner = synPkg.creator;
-        if (msg.sender != owner) {
-            require(hasRole(ROLE_CREATE, owner, msg.sender), "no permission to create");
+        address _sender = sender;
+        {
+            address _owner = synPkg.creator;
+            string memory _name = synPkg.name;
+            if (_sender != _owner) {
+                require(hasRole(ROLE_CREATE, _owner, _sender), "no permission to create");
+            }
+
+            // make sure the extra data is as expected
+            require(extraData.callbackData.length < maxCallbackDataLength, "callback data too long");
+            extraData.appAddress = _sender;
+            synPkg.extraData = abi.encode(extraData);
+
+            emit CreateSubmitted(_owner, _sender, _name);
         }
 
-        // make sure the extra data is as expected
-        require(extraData.callbackData.length < maxCallbackDataLength, "callback data too long");
-        extraData.appAddress = msg.sender;
-        synPkg.extraData = abi.encode(extraData);
-
-        ICrossChain(CROSS_CHAIN).sendSynPackage(
-            BUCKET_CHANNEL_ID,
-            abi.encodePacked(TYPE_CREATE, abi.encode(synPkg)),
-            relayFee,
-            _ackRelayFee
-        );
-
         // transfer all the fee to tokenHub
         (bool success, ) = TOKEN_HUB.call{ value: address(this).balance }("");
         require(success, "transfer to tokenHub failed");
 
```

Comment:
What is the difference between `createBucket` and `deleteBucket`?  Because I only see the change of `createBucket`, but no similar to `deleteBucket`, are't they both cross chain messasge?

Reply:
added prepareDeleteBucket